src/framework/theme/components/calendar-kit/components/calendar-navigation/calendar-view-mode.component.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { NbCalendarViewMode } from '../../model';
+import { NbCalendarYearModelService } from '../../services/calendar-year-model.service';
+import { NbDateService } from '../../services/date.service';
+import { TranslationWidth } from '@angular/common';
+
+
+@Component({
+  selector: 'nb-calendar-view-mode',
+  template: `
+    <button nbButton (click)="changeMode.emit()" ghost status="basic">
+      {{ getText() }}
+      <nb-icon [icon]="getIcon()" pack="nebular-essentials"></nb-icon>
+    </button>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NbCalendarViewModeComponent<D> {
+  @Input() date: D;
+  @Input() viewMode: NbCalendarViewMode = NbCalendarViewMode.DATE;
+  @Output() changeMode = new EventEmitter<void>(true);
+
+  constructor(
+    protected dateService: NbDateService<D>,
+    protected yearModelService: NbCalendarYearModelService<D>,
+  ) {}
+
+  getText(): string {
+    if (!this.date) {
+      return '';
+    }
+
+    switch (this.viewMode) {
```

Comment:
> we don't have validation for this.viewMode
We have type set on `viewMode`

> default value for that switch/case statement
Looks like silent fail. Why it's better?

Reply:
I agree with you, partially. If we don't have a default value, then, we ought to fire an exception or something, otherwise, we still have silent fail)